### ISSUE TITLE
Memory auto-detection fails on new OSX machines

### ISSUE THREAD

**Issue by s-andrews** [s-andrews/SeqMonk#257]
We use the output of:

```
top -l 1 -n 0
```

..on OSX to detect the amount of physical memory on the machine.  We had a report that this was failing and only giving default 1.5GB allocation on a machine with 32GB.  When the user ran the command they got:

```
Processes: 587 total, 2 running, 585 sleeping, 2965 threads 
2022/02/09 12:00:46
Load Avg: 2.88, 2.16, 2.09 
CPU usage: 1.72% user, 10.91% sys, 87.35% idle 
SharedLibs: 554M resident, 107M data, 128M linkedit.
MemRegions: 149884 total, 9010M resident, 715M private, 5797M shared.
PhysMem: 31G used (1858M wired), 223M unused.
VM: 221T vsize, 3780M framework vsize, 0(0) swapins, 0(0) swapouts.
Networks: packets: 5485405/7515M in, 600859/105M out.
Disks: 1830895/42G read, 793558/37G written.
```

The regex we use to find memory is:

```
^PhysMem:.*?(\d+)([MG])\s+used.*?(\d+)([MG])\s+(free|unused)
```

..and with the addition of the (1858M wired) it no longer matches this pattern so the detection fails.  We need to update the code to account for this variant of top output.

**Comment by s-andrews** [comment]
Actually this is a lie - it does work.  I build a small test script with this text in it and the output was:

Total mem is 31967

..which is the right value, so there must be something else going wrong on this system.

**Comment by s-andrews** [comment]
This was a spurious bug.  The actual (now fixed) bug was in #258
